editor/filesystem_dock.cpp

```diff
@@ -1411,14 +1411,28 @@ void FileSystemDock::_make_scene_confirm() {
 void FileSystemDock::_file_removed(String p_file) {
 	emit_signal("file_removed", p_file);
 
-	path = "res://";
+	// Find the closest parent directory available, in case multiple items were deleted along the same path.
+	path = p_file.get_base_dir();
+	DirAccess *da = DirAccess::create(DirAccess::ACCESS_RESOURCES);
```

Comment:
Use `DirAccessRef`, then you don't need to `memdelete()` it.

Reply:
Done!